### PR TITLE
feat(fdr): separate A32NX and A380X FDR files and converters; added additional data

### PR DESCRIPTION
**Pull request by aguther**
## Summary of Changes
This PR separates the FDR interfaces of A32NX and A380X and adds some base and aircraft specific data.

### A32NX
- 2 ELAC
- 3 SEC
- 2 FAC

### A380X
- 3 PRIM
- 3 SEC
- 2 Pseudo FAC

The interface numbering is now in a different format to also distinct between the two planes. It is now in the format `<PLANE 3-digit><INTERFACE 4-digit>`, example: `3800001` or `3200001`.

It also adds a setting in the EFB to hot-toggle the FDR. The default value can still be set in the corresponding INI file.

## Testing instructions
Just fly the airplane and test with the build of the FDR2CSV that resulting FDR files can be converted.

## How to download the PR for QA

Every new commit to this PR will cause new A32NX and A380X artifacts to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, find and click on the **PR Build** tab
1. Click on either **flybywire-aircraft-a320-neo** or **flybywire-aircraft-a380-842** download link at the bottom of the page

### CODE REVIEW

File: fbw-a32nx/src/wasm/fbw_a320/src/FlightDataRecorder.cpp [hunk 1]
```diff
@@ -51,6 +54,38 @@ void FlightDataRecorder::update(AutopilotStateMachine* autopilotStateMachine,
   manageFlightDataRecorderFiles();
 
   // write data to file
+
+  // ELACs
+  for (int i = 0; i < 2; ++i) {
+    auto bus_outputs = elacs[i].getBusOutputs();
+    fileStream->write((char*)(&bus_outputs), sizeof(bus_outputs));
```
Comment (tracernz) [inline fbw-a32nx/src/wasm/fbw_a320/src/FlightDataRecorder.cpp:61]: These casts are undefined behaviour (strict aliasing rule).

Comment (aguther) [reply fbw-a32nx/src/wasm/fbw_a320/src/FlightDataRecorder.cpp:61]: Do you have an idea how to improve this part of the code?

Comment (tracernz) [reply fbw-a32nx/src/wasm/fbw_a320/src/FlightDataRecorder.cpp:61]: Disregard, you actually can do this safely with char specifically so you're all good.

File: fbw-a32nx/src/wasm/fbw_a320/src/recording/FlightDataRecorder.cpp [hunk 2]
```diff
@@ -37,11 +37,14 @@ void FlightDataRecorder::initialize() {
   std::cout << "WASM: Flight Data Recorder Configuration : Interface Version              = " << INTERFACE_VERSION << std::endl;
 }
 
-void FlightDataRecorder::update(AutopilotStateMachine* autopilotStateMachine,
+void FlightDataRecorder::update(const BaseData& baseData,
+                                const AircraftSpecificData& aircraftSpecificData,
+                                Elac elacs[2],
```
Comment (lukecologne) [inline fbw-a32nx/src/wasm/fbw_a320/src/recording/FlightDataRecorder.cpp:42]: Looks like this passes by value, a reference might be more performant

**Review by tracernz — APPROVED** [review]
LGTM. Would be good for Luke to double check.

**Comment by Saschl** [comment]
Quality Assurance Tester Report

Discord Username  : saschl
Object of testing : #9125
Aircraft          : A32NX and A380X
Tier of Testing   : 1
Date              : 02/11/2024 

Testing Process: 
 
Spawned in A380X and A32NX and verified FDR files being generated.
Converted .fdr files for both planes to csv with fdr2csv utility.


Testing Results: 
Passed

**Pull request merged**